test/integration/etcd/server.go

```diff
@@ -62,6 +67,17 @@ AwEHoUQDQgAEH6cuzP8XuD5wal6wf9M6xDljTOPLX2i8uIp/C/ASqiIGUeeKQtX0
 /IR3qCXyThP/dbCiHrF3v1cuhBOHY8CLVg==
 -----END EC PRIVATE KEY-----`
 
+func registerEffectiveEmulationVersion(t *testing.T) {
```

Comment:
@liggitt: Unfortunately I couldn't do this as part of `configFuncs` as this must be called before we create ` options.NewServerRunOptions()`. It locks the registered version https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go#L104-L108. cc @siyuanfoundation

Reply:
ok, if it becomes conditional on an emulatedVersion passed into the server start method, which only the one test we care about sets, that's ok as well

Reply:
Yes, this is only called on the one test we care about: https://github.com/kubernetes/kubernetes/blob/c62628e520f576162e6836debd81f5c3ee767450/test/integration/etcd/etcd_storage_path_test.go#L95